src/hyperion/external_interaction/nexus/write_nexus.py

```diff
@@ -79,6 +77,9 @@ def create_nexus_file(self):
         initialised.
         """
         start_time = get_current_time()
+        est_end_time = get_predicted_end_time(
+            start_time, self.detector.exp_time * self.full_num_of_images
+        )
```

Comment:
Should: It's a bit of a faff to get current time, convert to string then convert back to get the end time. I think we could just do it in one function like `start_time, est_end_time = get_start_and_predicted_end_time(time_expected)`, which will be simpler

Reply:
Good point, done